spec/features/commands_spec.rb

```diff
@@ -534,6 +534,41 @@
     expect(action_response.message).to eq Slack::Message.new(text: "Oops! We had a problem running your command, but we've been notified")
   end
 
+  scenario 'github deployment does not start after 30 simulated secs and triggers watchdog' do
+    # make sure our queue is clear before starting test.
+    GithubDeploymentWatchdogWorker.clear
+
+    # our deployment watchdog worker should start with an empty queue.
+    expect(GithubDeploymentWatchdogWorker.jobs.size).to eq 0
+
+    expect(User).to receive(:find).and_return(users(:david))
+    expect(users(:david)).to receive(:slack_account_for_github_organization).and_return(slack_accounts(:david))
+
+    # simulate a slashdeploy invocation.
+    command '/deploy acme-inc/api to production', as: slack_accounts(:david)
+
+    expect(command_response).to be_in_channel
+    expect(deployment_requests).to eq [
+      [users(:david), DeploymentRequest.new(repository: 'acme-inc/api', ref: 'master', environment: 'production')]
+    ]
+
+    # our deployment watchdog worker should start with an empty queue.
+    expect(GithubDeploymentWatchdogWorker.jobs.size).to eq 1
+
+    # setup expectations of the worker notifying the user there was an issue.
+    expect(SlashDeploy.service).to receive(:direct_message).with(
+      slack_accounts(:david),
+      GithubNoDeploymentStatusMessage,
```

Comment:
We should be asserting the message content:

```ruby
expect(slack).to receive(:direct_message).with(
  slack_accounts(:david),
  Slack::Message.new(...)
)
```

It's somewhat verbose, but has saved me a bunch of times during refactors (we can cut down on the verbosity in the future by using something like gold master approvals).

Reply:
@ejholmes        `undefined local variable or method `slack' for #<RSpec::ExampleGroups::SlashCommands_2:0x007fdf72e66eb0>`